client/deviceCache.go

```diff
@@ -0,0 +1,209 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/plgd-dev/device/v2/client/core"
+	"github.com/plgd-dev/go-coap/v3/pkg/cache"
+	"go.uber.org/atomic"
+)
+
+type DeviceCache struct {
+	deviceExpiration time.Duration
+	devicesCache     *cache.Cache[string, *core.Device]
+	logger           core.Logger
+
+	closed atomic.Bool
+	done   chan struct{}
+}
+
+// Creates a new cache for devices.
+// - deviceExpiration: default expiration time for the device in the cache, 0 means infinite. The device expiration is refreshed by getting or updating the device.
+// - pollInterval: pool interval for cleaning expired devices from the cache
+// - logger: logger for logging
+func NewDeviceCache(deviceExpiration, pollInterval time.Duration, logger core.Logger) *DeviceCache {
+	done := make(chan struct{})
+	cache := cache.NewCache[string, *core.Device]()
+	if deviceExpiration > 0 {
```

Comment:
Shouldn't `deviceExpiration` be used somewhere, otherwise it can be just a bool instead of a duration..